app/GRAPHQL/components/GqlForm/GQLForm.tsx

```diff
@@ -0,0 +1,254 @@
+    }
+
+    if (errors.query && errMessages.includes(errors.query)) {
+      updatedErrors.query = t('errInvalidGraphQLQuery');
+    }
+
+    if (errors.endpointSDL && errMessages.includes(errors.endpointSDL)) {
+      updatedErrors.endpointSDL = t('errInvalidUrl');
+    }
+
+    if (errors.variables && errMessages.includes(errors.variables)) {
+      updatedErrors.variables = t('errInvalidJson');
+    }
+
+    setErrors(updatedErrors);
+  }, [t]);
+
+  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    const form = event.currentTarget;
+    const data = new FormData(form);
+
+    const formData = {
+      endpointURL,
+      endpointSDL,
+      query: queryArea,
+      variables: variablesArea || null,
+    };
+
+    try {
+      await gqlFormSchema.validate(formData, { abortEarly: false });
+      setErrors({});
+      saveRequestToLS({
+        method: 'GRAPHQL',
+        generatedURL: pathname + '?' + search,
+        EndpointURL: endpointURL,
+      });
+      action(data);
+    } catch (validationError) {
+      if (validationError instanceof Yup.ValidationError) {
+        const formErrors = validationError.inner.reduce(
+          (acc, err) => ({
+            ...acc,
+            [err.path!]: err.message,
+          }),
+          {}
+        );
+        setErrors(formErrors);
+      }
+    }
+  };
+
+  const formatQuery = async () => {
+    if (ref.current) {
+      const formData = {
+        query: queryArea,
+        variables: variablesArea || null,
+      };
+
+      try {
+        await prettifySchema.validate(formData, { abortEarly: false });
+        setErrors({});
+        if (queryArea) {
+          setQueryArea(gqlPrettier(queryArea));
+        }
+
+        if (variablesArea) {
+          setVariablesArea(JSON.stringify(JSON.parse(variablesArea), null, 2));
+        }
+      } catch (validationError) {
+        if (validationError instanceof Yup.ValidationError) {
+          const formErrors = validationError.inner.reduce(
+            (acc, err) => ({
+              ...acc,
+              [err.path!]: err.message,
+            }),
+            {}
+          );
+          setErrors(formErrors);
+        }
+      }
+    }
+  };
+
+  return (
+    <div className={styles.graphiqlContainer}>
+      {open && show && (
+        <article className={styles.docsContainer}>
+          {
+            <SchemaDocumentation
+              valueSDL={endpointSDL ? endpointSDL : endpointURL}
+            />
+          }
+        </article>
+      )}
+      <form
+        className={styles.form}
+        ref={ref}
+        action={action}
+        onSubmit={handleSubmit}
+        noValidate
+      >
+        <EndpointURL
+          setURL={setEndpointURL}
+          urlValue={endpointURL}
+          setOpen={setOpen}
+        />
+        {errors.endpointURL && (
+          <p className={styles.errorText}>{errors.endpointURL}</p>
+        )}
```

Comment:
Looks like this can be moved to the separate component, which can be reused in several places.

Reply:
In EndPointURL and EndPointSDL significantly different logic inside those components. I can't refactor in normal way :(